opensrp-chw/src/main/java/org/smartregister/chw/interactor/FpUpcomingServicesInteractor.java

```diff
@@ -2,21 +2,139 @@
 
 import android.content.Context;
 
+import org.jeasy.rules.api.Rules;
+import org.jetbrains.annotations.Nullable;
+import org.joda.time.DateTime;
+import org.joda.time.format.DateTimeFormat;
+import org.smartregister.chw.R;
 import org.smartregister.chw.anc.domain.MemberObject;
+import org.smartregister.chw.anc.domain.Visit;
 import org.smartregister.chw.anc.interactor.BaseAncUpcomingServicesInteractor;
 import org.smartregister.chw.anc.model.BaseUpcomingService;
+import org.smartregister.chw.core.rule.FpAlertRule;
+import org.smartregister.chw.core.utils.CoreConstants;
+import org.smartregister.chw.core.utils.FpUtil;
+import org.smartregister.chw.core.utils.HomeVisitUtil;
+import org.smartregister.chw.core.utils.VaccineScheduleUtil;
+import org.smartregister.chw.fp.dao.FpDao;
+import org.smartregister.chw.fp.domain.FpAlertObject;
+import org.smartregister.chw.fp.util.FamilyPlanningConstants;
 
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.Date;
 import java.util.List;
 
 public class FpUpcomingServicesInteractor extends BaseAncUpcomingServicesInteractor {
-    private Flavor flavor = new FpUpcomingServicesInteractorFlv();
+    protected MemberObject memberObject;
+    protected Context context;
 
     @Override
-    protected List<BaseUpcomingService> getMemberServices(Context context, MemberObject memberObject) {
-        return flavor.getMemberServices(context, memberObject);
+    public List<BaseUpcomingService> getMemberServices(Context context, MemberObject memberObject) {
+        this.memberObject = memberObject;
+        this.context = context;
+        VaccineScheduleUtil.updateOfflineAlerts(memberObject.getBaseEntityId(), new DateTime(memberObject.getDob()), CoreConstants.SERVICE_GROUPS.CHILD);
+        List<BaseUpcomingService> serviceList = new ArrayList<>();
+        evaluateFp(serviceList);
+        return serviceList;
     }
 
-    public interface Flavor {
-        List<BaseUpcomingService> getMemberServices(Context context, MemberObject memberObject);
+    private void evaluateFp(List<BaseUpcomingService> serviceList) {
+        String fpMethod = null;
+        String fp_date = null;
+        Integer fp_pillCycles = null;
+        Rules rule = null;
+        Integer count = null;
+        Date serviceDueDate = null;
+        Date serviceOverDueDate = null;
+        String serviceName = null;
+        List<FpAlertObject> familyPlanningList = FpDao.getFpDetails(memberObject.getBaseEntityId());
+        if (familyPlanningList.size() > 0) {
+            for (FpAlertObject familyPlanning : familyPlanningList) {
+                fpMethod =familyPlanning.getFpMethod();
+                fp_date = familyPlanning.getFpStartDate();
+                fp_pillCycles = familyPlanning.getFpPillCycles();
+                rule = FpUtil.getFpRules(fpMethod);
+            }
+        }
+        fpMethod = getTranslatedValue(fpMethod);
```

Comment:
It looks like `fpMethod` translated value is being used in getting the latest visit and visit status queries. Are translated values being saved in the DB?
I was under the impression string translations are mostly needed when displaying text and not querying data :thinking:

Reply:
Fixed.